shark_turbine/kernel/wave/wave.py

```diff
@@ -209,11 +218,66 @@ def _trace_and_get_kernel_signature(
         if kwargs.get("canonicalize", False):
             canonicalize_module(mb.module_op)
 
-        return mb, graph
+        return mb, graph, exe, kernel_sig, entrypoint_name
 
     def test_execute(self, args, kwargs):
         # For now only tracing
-        mb, graph = self._trace_and_get_kernel_signature(args, kwargs)
+        (
+            mb,
+            graph,
+            exe,
+            kernel_sig,
+            entrypoint_name,
+        ) = self._trace_and_get_kernel_signature(args, kwargs)
+
+        if kwargs.get("run", False):
+            # TODO: cache compiled code
+            host_codegen.isolated_test_call(mb, exe, kernel_sig, entrypoint_name)
+            asm = mb.module_op.get_asm()
+
+            kernel_inputs = []
+            kernel_outputs = []
+            for arg, b in zip(args, kernel_sig.kernel_buffer_bindings):
+                usage = b.kernel_buffer_type.usage
+                if usage == kernel_codegen.KernelBufferUsage.INPUT:
+                    kernel_inputs.append(arg)
+
+                if usage == kernel_codegen.KernelBufferUsage.OUTPUT:
+                    kernel_outputs.append(arg)
+
+            # TODO: Unhardcode
+            target = "gfx942"
```

Comment:
What do you think of a "configuration" class -  which contains things like device(rocm/hip) target arch(gfx942), whether to print mlir after all, etc.? We can instantiate that dict and pass it in as an argument.

Reply:
Good idea, Done.

Reply:
thanks!